blueprints/test-framework-detector.js

```diff
@@ -12,7 +14,12 @@ module.exports = function(blueprint) {
     if ('ember-cli-qunit' in dependencies) {
       type = 'qunit';
     } else if ('ember-cli-mocha' in dependencies) {
-      type = 'mocha';
+      var checker = new VersionChecker({ project: this.project });
+      if (fs.existsSync(this.path + '/mocha-0.12-files') && checker.for('ember-cli-mocha', 'npm').satisfies('>=0.12.0')) {
```

Comment:
@pangratz that check allows us to have some blueprints that care about the Mocha version and others that don't. This is not technically needed in this case, but I've reused the code from https://github.com/emberjs/ember.js/pull/14670 to reduce differences.